mlir/include/mlir/Transforms/InliningUtils.h

```diff
@@ -253,13 +254,15 @@ class InlinerInterface
 /// provided, will be used to update the inlined operations' location
 /// information. 'shouldCloneInlinedRegion' corresponds to whether the source
 /// region should be cloned into the 'inlinePoint' or spliced directly.
-LogicalResult inlineRegion(InlinerInterface &interface, Region *src,
+LogicalResult inlineRegion(InlinerInterface &interface,
+                           const InlinerConfig &config, Region *src,
```

Comment:
Requiring the entire config seems overkill to me. We should be able to just take a `function_ref` that has the signature of the callback here.